File: 1704 - Determine if String Halves Are Alike/1704.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        int length = len(s)
        vowels = ['a','e','i','o','u']
        vowelsCountFirst = len([vowel for char in s[:length//2] if char in vowels])
        vowelsCountSecond = len([vowel for char in s[length//2:] if char in vowels])
        return vowelsCountFirst == vowelsCountSecond

File: 1704 - Determine if String Halves Are Alike/1704.cpp
class Solution {
    bool checkVowel(char ch) {
        if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u')
            return true;
        if (ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U')
            return true;
        return false;
    }
public:
    bool halvesAreAlike(string s) {
        int length = s.size();
        
        int vowelsCountFirst = 0, vowelsCountSecond = 0;
        
        for (int i = 0, j = length/2; i < length/2; i++) {
            
            if ( checkVowel( s[i] ) )
                vowelsCountFirst++;
            
            if ( checkVowel( s[j+i]) )
                vowelsCountSecond++;
                
        }
        
        return vowelsCountFirst == vowelsCountSecond;
    }
};